cloud_functions/data_processing/src/methods/subtract_geometries.py

```diff
@@ -0,0 +1,120 @@
+import pandas as pd
+import geopandas as gpd
+from joblib import Parallel, delayed
+from tqdm.auto import tqdm
+from src.utils.gcp import (
+    read_json_df,       # Reads a .json or .geojson file from GCS and returns a DataFrame or GeoDataFrame.
+    upload_gdf_zip      # Saves a GeoDataFrame to GCS as a .geojson file.
+)
+from src.utils.logger import Logger
+
+logger = Logger()
+
+def process_country(country: str, 
+                    boundary_gdf: gpd.GeoDataFrame, 
+                    pa_gdf: gpd.GeoDataFrame
+    ):
+    """
+    Subtracts the protected area from the total area and returns a GeoDataFrame.
+
+    Parameters
+    ----------
+    country : str 
+        Country name (3-letter abbreviation).
+    boundary_gdf : gpd.GeoDataFrame
+        Total areas GeoDataFrame.
+    pa_gdf : gpd.GeoDataFrame
+        Protected areas GeoDataFrame.
+    
+    Returns
+    -------
+    gpd.GeoDataFrame
+        GeoDataFrame of boundary_gdf with pa_gdf subtracted, retaining the original fields of boundary_gdf.
+    """
+    try:
+        country_area = boundary_gdf[boundary_gdf['location'] == country]
+        country_pa = pa_gdf[pa_gdf['ISO3'].str.contains(country)].dissolve()
+        if country_pa.empty:
+            # If no protected areas, return original boundary
+            return country_area
+        else:
+            # If protected areas found, return original boundary with protected areas removed
+            return country_area.overlay(country_pa, how='difference')
+    
+    except Exception as e:
+        logger.warning({'message': f'Error processing {country}: {e}'})
+        return None
+
+def update_total_area_minus_pa(bucket: str, 
```

Comment:
nit: The general convention we've established is that `update_` for a function name indicates that the data will be updated in the database. `generate_` is what's been used for the functions that create the dataset and store it in GCS for future "updating"


```suggestion
def generate_total_area_minus_pa(bucket: str, 
```